src/linagora.esn.unifiedinbox/app/controllers.js

```diff
@@ -169,6 +169,8 @@ require('./services/common/inbox-utils.service.js');
         return inboxJmapHelper
           .getMessageById(emailId)
           .then(function(message) {
+            inboxMessagesCache[emailId] = message;
```

Comment:
there is no risk of leaking memory here?

Reply:
The worst case you could have every loaded messages stored in the cache, but you cannot have duplicate messages (because they would have the same id) I think, so yes you are probably right.

Reply:
For the mailbox cache you just want to get the list of mailboxes, but for the message cache, you want to access a message by its id.

Reply:
ahahah, the cache eviction. Thank you Raphael for your vigilance ! ;-)

Reply:
> why inboxMessagesCache is defined in a different code style than inboxMailboxesCache
*inboxMessagesCache* style is the good one in JS, you don't need to loop all the array items to get your item. 
Such as in java you have the choice between array or Map, so you consider {} as a Map and you can get or delete an item without a loop.